Form/DataTransformer/MoneyToLocalizedStringTransformer.php

```diff
@@ -21,18 +21,19 @@ class MoneyToLocalizedStringTransformer extends NumberToLocalizedStringTransform
 {
     /** @var string ISO currency code **/
     private $currencyCode;
-    /** @var \Money\Parser\DecimalMoneyParser **/
-    private $moneyParser;
+    /** @var \Money\Currencies|null */
+    private $currencies;
 
     /**
-     * @param string $currencyCode ISO currency code
-     * @param int    $scale
-     * @param bool   $grouping
+     * @param string                 $currencyCode ISO currency code
+     * @param int                    $scale
+     * @param bool                   $grouping
+     * @param \Money\Currencies|null $currencies
      */
-    public function __construct($currencyCode, $scale, $grouping)
+    public function __construct($currencyCode, $scale, $grouping, Currencies $currencies = null)
     {
-        $this->moneyParser = new DecimalMoneyParser(new ISOCurrencies());
         $this->currencyCode = $currencyCode;
+        $this->currencies = $currencies ? $currencies : new ISOCurrencies();
```

Comment:
`$this->currencies = $currencies ?: new ISOCurrencies();`

Reply:
Or it might be better to simply require a `MoneyParser` and `MoneyFormatter` in constructor instead. That way the transformer would not be locked to the `DecimalMoneyParser` and `DecimalMoneyFormatter`. What do you think @kucharovic?

Reply:
@dadscript can you update it according @enumag's description?

Reply:
How is it able to do that and don't break BC?

Reply:
@dadscript you're right – it would be BC. I'll do that in new minor version. Please, simplify that ternary operator, and I'll merge it.

Reply:
fixed